rare/components/tabs/games/import_sync/import_group.py

```diff
@@ -146,38 +149,63 @@ def find_app_name(self, path):
             return None
 
     def import_game(self, path=None):
-        app_name = self.app_name.text()
-        if not path:
-            path = self.path_edit.text()
-        if not app_name:
-            # try to find app name
-            if a_n := self.find_app_name(path):
-                app_name = a_n
-            else:
-                self.info_label.setText(self.tr("Could not find app name"))
-                return
-
-        if not (
-            err := legendary_utils.import_game(self.core, app_name=app_name, path=path)
-        ):
-            igame = self.core.get_installed_game(app_name)
-            self.info_label.setText(
-                self.tr("Successfully imported {}").format(igame.title)
-            )
-            self.app_name.setText(str())
-            self.signals.update_gamelist.emit([app_name])
-
-            if (
-                igame.version
-                != self.core.get_asset(app_name, igame.platform, False).build_version
-            ):
-                # update available
-                self.signals.add_download.emit(igame.app_name)
-                self.signals.update_download_tab_text.emit()
-
-        else:
-            logger.warning(f'Failed to import "{app_name}"')
-            self.info_label.setText(
-                self.tr("Could not import {}: {}").format(app_name, err)
-            )
-            return
+        @dataclass
+        class Arguments:
+            app_path: str
+            app_name: str
+            platform: str
+            disable_check: bool
+            skip_dlcs: bool
+            with_dlcs: bool
+            yes: bool
+            log_dest: QLabel
+        args = Arguments(
+            app_path=self.path_edit.text(),
+            app_name=self.app_name.text(),
+            platform='Windows',
+            disable_check=False,
+            skip_dlcs=False,
+            with_dlcs=False,
+            yes=False,
+            log_dest=self.info_label,
+        )
```

Comment:
Seems like a perfect usecase for named tuples here, if legendary doesn't require `args` to be a conventional class.

Reply:
I didn't figure out what `args` actually is to replicate it, and named tuple might fit the bill just right. This part was a quick way for me to test if the shim worked correctly and show-case it.

Reply:
Decided to use `Namespace` from `argparse`